UnityProject/Assets/Scripts/Items/Weapons/MagazineBehaviour.cs

```diff
@@ -270,6 +273,21 @@ public virtual String Examine(Vector3 pos)
 	{
 		return "Accepts " + ammoType + " rounds (" + (ServerAmmoRemains > 0 ? (ServerAmmoRemains.ToString() + " left") : "empty") + ")";
 	}
+
+	#if UNITY_EDITOR
+	[CustomEditor(typeof(MagazineBehaviour), true)]
+	public class MagEditor : Editor
+	{
+		public override void OnInspectorGUI()
+		{
+			DrawDefaultInspector(); // for other non-HideInInspector fields
+
+			MagazineBehaviour script = (MagazineBehaviour)target;
+
+			script.isClip = EditorGUILayout.Toggle("isClip", script.isClip);
```

Comment:
what is this for?

Reply:
this makes the hidden field isClip show, then the child scripts reset the inspector to default hiding that field as using it in them would break some things

Reply:
I feel like using ``ShowIf`` from NaughtyAttributes would be cleaner. 

Another way to see it is, why do the children classes need this field if they won't use it? You could delete from the super class and only add it to the children classes that make use of it.

Reply:
ive never managed to get showif to work
the superclass is the only thing to make use of isClip

Reply:
```cs
[HideInInspector]
public bool isClip = false;
```

Where do you change the value of this? Why is it public?

Reply:
in the inspector.
it is public because gun.cs needs reference to it

Reply:
> I feel like using `ShowIf` from NaughtyAttributes would be cleaner.
> 
> Another way to see it is, why do the children classes need this field if they won't use it? You could delete from the super class and only add it to the children classes that make use of it.

isCell is used only by the eletrical magazine script, however moving it to the child script would result in a large quantity of unavoidable copy pasting, which is why it is in the super script in the first place. to avoid code duplication. the bool itself is atuomatically toggled by the child script. to avoid accidentally toggling it, which would break some behaviour.

Reply:
Can a weapon be cell and clip? If they can’t, then this shouldn’t be two different checkbox but a drop down with both options

Reply:
> Can a weapon be cell and clip? If they can’t, then this shouldn’t be two different checkbox but a drop down with both options

aight, gimme a bit to implement that then